cmd/main.go

```diff
@@ -78,12 +84,50 @@ func performCommand(cmd *cobra.Command, args []string) error {
 		return nil
 	}
 
+	if config.AutoCert != "" && (config.Cert != "" || config.Key != "") {
+		return errors.New("can't specify auto-cert and key/cert")
+	}
+
 	t := server.NewTracker(config.PoolSize)
 
+	m, err := getLetsEncryptManager()
+	if err != nil {
+		return err
+	}
+
 	// enable stats if port specified
 	if config.StatsPort > 0 {
-		go server.StartStatsServer(config.StatsPort, config.Cert, config.Key, t)
+		go server.StartStatsServer(config.StatsPort, config.Cert, config.Key, t, m)
+	}
+
+	return server.Start(config.Port, config.Cert, config.Key, config.Debug, t, m)
+}
+
+func getLetsEncryptManager() (*autocert.Manager, error) {
+	configDir, err := config.configDir()
+	if err != nil {
+		return nil, err
+	}
+
+	certDir := filepath.Join(configDir, "certs")
+	if _, err := os.Stat(certDir); os.IsNotExist(err) {
+		err = os.MkdirAll(certDir, 0755)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	if config.AutoCert != "" {
+		m := &autocert.Manager{
+			Cache:      autocert.DirCache(certDir),
+			Prompt:     autocert.AcceptTOS,
+			HostPolicy: autocert.HostWhitelist(config.AutoCert),
+		}
+
+		go http.ListenAndServe(":http", m.HTTPHandler(nil))
+
+		return m, err
```

Comment:
Nit: `return m, nil` - `err` should always be nil here

Reply:
Fixed.